.github/workflows/release-embedding-sdk.yml

```diff
@@ -4,32 +4,87 @@ on:
   workflow_dispatch:
     inputs:
       # TODO: Add a version validation, so the workflow won't fail when publishing to npm
-      sdk_version:
-        description: "SDK version (e.g. 0.1.3)"
-        type: string
-        required: true
-      git_ref:
-        description: "A git commit hash that will be used to build the SDK"
-        type: string
+      branch:
+        description: "Branch we want to release the SDK from"
+        type: choice
         required: true
+        default: "master"
+        options:
+          - master
+          - release-x.51.x
 
 concurrency:
-  # We want to ensure only one job is running at a time because
+  # We want to ensure only one job is running at a time per branch because
   # there could be a conflict when updating the readme file.
-  group: ${{ github.workflow }}
+  group: ${{ github.workflow }}-${{ inputs.branch }}
   cancel-in-progress: true
 
 jobs:
+  workflow-summary:
+    name: Log inputs
+    runs-on: ubuntu-22.04
+    timeout-minutes: 5
+    steps:
+      - name: Generate workflow summary
+        run: |
+          echo '**Inputs:**' >> $GITHUB_STEP_SUMMARY
+          echo '' >> $GITHUB_STEP_SUMMARY
+          echo '- `branch`: ${{ inputs.branch }}' >> $GITHUB_STEP_SUMMARY
+          echo '' >> $GITHUB_STEP_SUMMARY
+          echo 'triggered by: @${{ github.event.sender.login }}' >> $GITHUB_STEP_SUMMARY
+
+  determine-version:
+    name: Determine SDK version
+    runs-on: ubuntu-22.04
+    timeout-minutes: 20
+    outputs:
+      sdk_version: ${{ steps.new-sdk-version.outputs.result }}
+    steps:
+      - name: Check out the code using the provided branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.branch }}
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Get current SDK version
+        id: current-sdk-version
+        run: |
+          VERSION=$(cat ./enterprise/frontend/src/embedding-sdk/package.template.json | jq -r .version)
+          echo "::set-output name=sdk_current_version::$VERSION"
+
+      - name: Get next SDK patch version
+        id: new-sdk-version
+        uses: actions/github-script@v7
+        with:
+          result-encoding: string
+          script: |
+            const currentVersion = "${{ steps.current-sdk-version.outputs.sdk_current_version }}";
+            const versionParts = currentVersion.split(".");
+            versionParts[versionParts.length - 1] = parseInt(versionParts.at(-1)) + 1;
+            const newVersion = versionParts.join(".");
+            console.log({currentVersion, newVersion});
+
+            return newVersion
+
+      - name: Append workflow summary
```

Comment:
This step logs this to the summary
![image](https://github.com/user-attachments/assets/8da8b2aa-9bb6-434c-9147-e0060c52eda4)